### PR TITLE
Add error styling to text input.

### PR DESCRIPTION
**Pull request by jkinkead**
Add "number" input type to input styling.

Errors!

### CODE REVIEW

File: less/defaults.less [hunk 1]
```diff
@@ -263,12 +264,21 @@ textarea {
...
 
 input[type="text"],
+input[type="number"],
 input[type="search"],
 input[type="password"],
 textarea {
   border: 1px solid @light-gray;
   box-shadow: inset 1px 1px 4px 0 @light-gray;
   width: 100%;
+  &.error {
```
Comment (codeviking) [inline less/defaults.less]: I recently read trello's CSS guide and have been convinced of the `mod-` and `is-` prefixes.  What do you think about calling this `mod-has-error` or `is-invalid`?

**Comment by codeviking** [comment]
Just one comment on using a `mod` or `is` prefix, if you're curious re the original source of this recommendation:

https://gist.github.com/bobbygrace/9e961e8982f42eb91b80?utm_source=html5weekly&utm_medium=email

I'll likely be proposing that we incorporate some of this into our style-guide.

File: less/defaults.less (hunk 1 above)
Comment (jkinkead) [inline less/defaults.less]: I think that's a cool idea, but I think we should adopt it all at once, not piecemeal. We've got a bunch of other class names (`clickable`, `hidden`, `visuallyhidden`) that don't follow that guideline, and it's a lot nicer to have one style of names. Sound good?

**Comment by jkinkead** [comment]
100% for updating the style guide, but I think we should update all of syrup at once.

**Comment by codeviking** [comment]
I'd opt for progressive edits -- it means less work in the future to port to an updated version.   

Also, existing class names like `clickable` and `hidden` are still a bit more clear to me than `error` -- `error` could be an error message, or an error state -- it's ambiguous.   So let's figure out a name that more directly represents it's intention, personally I like `mod-has-error`.  If you don't want to adopt the `mod` prefix yet let's go with `has-error` at the very least?

**Comment by jkinkead** [comment]
`has-error` sounds awesome! I'll update to that. :)

Thanks for the review!

**Comment by jkinkead** [comment]
I'm going to try to publish the pages now - I'll let you handle any library updates.

**Comment by jkinkead** [comment]
. . . and it's deployed!

**Pull request merged**